test/client_test.rb

```diff
@@ -374,73 +374,4 @@
       refute_same $client.connection, client.connection
     end
   end
-
-  describe "automatic retry of requests" do
-    before do
-      @events = []
-      @subscriber = ActiveSupport::Notifications.subscribe do |*args|
-        @events << ActiveSupport::Notifications::Event.new(*args)
-      end
-    end
-
-    after do
-      @events.clear
-      ActiveSupport::Notifications.unsubscribe(@subscriber)
-    end
-
-    it "defaults to no retries" do
-      stub_request(:get, $client.url+"/_cat/indices").
-        to_timeout.then.
-        to_return({
-          headers: {"Content-Type" => "text/plain; charset=UTF-8"},
-          body: "green open test-index 1 0 0 0 159b 159b"
-        })
-
-      assert_raises(ElastomerClient::Client::ConnectionFailed) {
-        $client.get("/_cat/indices")
-      }
-    end
-
-    it "retries up to `max_retries` times" do
-      stub_request(:get, $client.url+"/test-index/_settings").
-        to_timeout.then.
-        to_timeout.then.
-        to_return({body: %q/{"acknowledged": true}/})
-
-      response = $client.index("test-index").settings(max_retries: 2)
-
-      assert_equal 2, @events.first.payload[:retries]
-      assert_equal({"acknowledged" => true}, response)
-    end
```

Comment:
If not too much work, I think it would be good to turn this test into an example of how to add retries to a client using the retry middleware in Faraday. That would require a different client instance since this one is configured without retries already.

What do you think?

Reply:
Also, we should probably update the README with an example of adding retries. I don't think accepting a block on initialize is mentioned in the README at all (sorry about that), so an example of using it would be helpful documentation.

Reply:
Great idea on both ends! I'll add those today

Reply:
Added!